shared/engine/index.js

```diff
@@ -248,36 +273,35 @@ class Engine {
     return wrappedResponse
   }
 
-  _generalIncomingRpc (method, param, response) {
+  _generalIncomingRpc (method: MethodKey, param: Object, response: any) {
     const listener = this.generalListeners[method]
 
     if (!listener) {
       return
     }
 
     // make wrapper so we only call this once
-    const wrappedResponse = this._wrapResponseOnceOnly(method, param, response)
+    const wrappedResponse = this._wrapResponseOnceOnly(method, param, response, null)
```

Comment:
lets require a waiting handler to this helper method so we don't have a waitinghandler?: ?WaitingHandlerType style declaration